src/deletepy/utils/logging_utils.py

```
@@ -0,0 +1,276 @@
+def log_operation(operation: str, user_id: str = None, **kwargs):
+    """Decorator for logging operation start/end with context.
+    
+    Args:
+        operation: Operation name
+        user_id: Optional user ID context
+        **kwargs: Additional context variables
+    """
+    def decorator(func):
+        def wrapper(*args, **func_kwargs):
+            logger = get_logger(func.__module__)
+            
```

Ruff (0.11.9)

Blank line contains whitespace

Remove whitespace from blank line

(W293)